src/convert.jl

```diff
@@ -213,6 +213,85 @@ end
 convert(::Type{HPolytope}, P::HPolytope) = P
 convert(::Type{VPolytope}, P::VPolytope) = P
 
+"""
+    convert(::Type{HPolytope}, Z::Zonotope{N}) where {N<:Real}
+
+Convert from zonotope to polytope in H-representation.
+
+### Input
+
+- `type` -- target type
+- `Z`    -- source zonotope
+
+### Output
+
+The zonotope represented as polytope in H-representation.
+
+### Notes
+
+The algorithm assumes that no generator is redundant.
+The result has ``2 \binom{p}{n-1}`` (with ``p`` being the number of generators
+and ``n`` being the ambient dimension) constraints, which is optimal under this
+assumption.
+
+### Algorithm
+
+We follow the algorithm presented in *Althoff, Stursberg, Buss: Computing
+Reachable Sets of Hybrid Systems Using a Combination of Zonotopes and Polytopes.
+2009.*
+
+The one-dimensional case is not covered by that algorithm; we manually handle
+this case, assuming that there is only one generator.
+"""
+function convert(::Type{HPolytope}, Z::Zonotope{N}) where {N<:Real}
+    p = ngens(Z)
+    n = dim(Z)
+    if p < n
+        error("can only convert a zonotope of order >= 1")
+    end
+
+    G = Z.generators
+    m = binomial(p, n - 1)
+    constraints = Vector{LinearConstraint{N}}(undef, 2 * m)
+
+    # special handling of 1D case
+    if n == 1
+        if p > 1
+            error("1D-zonotope conversion only supports a single generator")
+        end
+
+        c = Z.center[1]
+        g = G[:, 1][1]
+        constraints[1] = LinearConstraint([N(1)], c + g)
+        constraints[2] = LinearConstraint([N(-1)], g - c)
+        return HPolytope(constraints)
+    end
+
+    @assert N <: AbstractFloat "cannot convert rational zonotopes"
```

Comment:
Why would you do this assertion here instead of dispatching on `AbstractFloat`?

Reply:
Because the case `n == 1` can be handled.

Reply:
Then the default conversion can be used in general (from `master`):

```julia
# convert a 2D zonotope
julia> Z = Zonotope(Rational{BigInt}[1, 0.0], Rational{BigInt}[1 2; 0 1])
Zonotope{Rational{BigInt}}(Rational{BigInt}[1//1, 0//1], Rational{BigInt}[1//1 2//1; 0//1 1//1])

julia> convert(HPolytope, Z)
HPolytope{Rational{Int64}}(LazySets.HalfSpace{Rational{Int64}}[HalfSpace{Rational{Int64}}(Rational{Int64}[0//1, 1//54], 1//54), HalfSpace{Rational{Int64}}(Rational{Int64}[1//1, -2//1], 2//1), HalfSpace{Rational{Int64}}(Rational{Int64}[-1//55, 2//55], 0//1), HalfSpace{Rational{Int64}}(Rational{Int64}[0//1, -1//2], 1//2)])
```

Idea:

```julia
function convert(::Type{HPolytope}, Z::Zonotope{N}) where {N<:AbstractFloat}
  ... # keep current method (without the assertion) since we dispatch on AbstractFloat
end

function convert(::Type{HPolytope}, Z::Zonotope{N}) where {N<:Real}
     # invoke default convertion or write it
end
```

Reply:
No, we want to get rid of `convert` because of #932. What we could do is take the detour over the v-rep.